documentation/api/query/v4/producers.markdown

```diff
@@ -0,0 +1,102 @@
+---
+title: "PuppetDB 4.1: Producers endpoint"
+layout: default
+canonical: "/puppetdb/latest/api/query/v4/producers.html"
+---
+
+[curl]: ../curl.html#using-curl-from-localhost-non-sslhttp
+[paging]: ./paging.html
+[query]: ./query.html
+[subqueries]: ./ast.html#subquery-operators
+[factsets]: ./factsets.html
+[reports]: ./reports.html
+[catalogs]: ./catalogs.html
+
+Producers are the Puppet masters that send reports, catalogs, and factsets to PuppetDB.
+
+When PuppetDB stores a report, catalog, or factset, it keeps track of the producer
+of the report/catalog/factset. PuppetDB also keeps a list of producers it has seen.
+You can query this list by making an HTTP request to the `/producers` endpoint.
+
+## `/pdb/query/v4/producers`
+
+This will return all producers known to PuppetDB.
+
+### URL parameters
+
+* `query`: optional. A JSON array containing the query in prefix notation. If
+  not provided, all results will be returned. See the sections below for the
+  supported operators and fields. For general info about queries,
+  see [our guide to query structure.][query]
+
+### Query operators
+
+See [the AST query language page](./ast.html)
+
+### Query fields
+
+* `"name"` (string): the certname of a producer.
```

Comment:
D: I'm just following whatever was done for environments. Should I change both or neither or??